### PR TITLE
Upgrade apollo libraries in Navigator

### PR DESCRIPTION
**Pull request by cocreature**
Apologies for the relatively large PR, I couldn’t figure out a way to make this smaller. Happy to hop on a call and walk someone through this if that is easier.

We don’t really have tests for the Navigator frontend at this point (we have some bitrotten ones in the repo). I’d like to change that but that seems out of scope for this PR. I ran through the full Java quickstart which is the most comprehensive Navigator workflow I’m aware off.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: navigator/frontend/src/applets/app/index.tsx [hunk 1]
```diff
@@ -29,7 +27,6 @@ export type Action
   | { type: 'TO_WATCHER', action: LedgerWatcher.Action }
   | { type: 'TO_CONFIG', action: ConfigSource.Action }
   | { type: 'RESET_APP' }
-  | ApolloAction
```
Comment (cocreature) [inline navigator/frontend/src/applets/app/index.tsx:32]: There is no longer any redux integration so we need to keep the two separate.

File: navigator/frontend/src/applets/contract/ContractComponent.tsx [hunk 2]
```diff
@@ -52,8 +52,10 @@ interface ActiveLinkProps extends OwnProps {
   isActive: boolean;
 }
 
-const ActiveLink = (props: ActiveLinkProps) => (
-  <Link {...props} />
+const ActiveLink: React.FC<ActiveLinkProps> = (props) => (
```
Comment (cocreature) [inline navigator/frontend/src/applets/contract/ContractComponent.tsx:55]: Without this, we get an error below where we try to do something like `<ActiveLink>…</ActiveLink>`. Not quite sure what got stricter but seems sensible to me.

**Review by garyverhaegen-da — APPROVED** [review]
Huge thanks for doing this, I can't imagine it being much fun.

**Comment by rautenrieth-da** [comment]
Thanks for keeping the Navigator code base up to date. 🙇‍♂️

**Pull request merged**